src/scripts/main.js

```diff
@@ -1,3 +1,179 @@
 'use strict';
 
-// write your code here
+const messageStart = document.querySelector('.message-start');
+const messageLoose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+
+const button = document.querySelector('.button');
+const gameScore = document.querySelector('.game-score');
+const boardRows = document.querySelectorAll('.field-row');
+
+let score = 0;
+let gameBoard = boardInit();
+
+function boardInit() {
+  return [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+}
+
+function resetGame() {
+  gameBoard = boardInit();
+  score = 0;
+  messageLoose.classList = 'message message-lose hidden';
+  messageWin.classList = 'message message-win hidden';
+}
+
+function renderGameBoard() {
+  for (let row = 0; row < 4; row++) {
+    for (let col = 0; col < 4; col++) {
+      boardRows[row].children[col].innerText
+        = gameBoard[row][col] ? gameBoard[row][col] : '';
+
+      boardRows[row].children[col].className
+        = `field-cell field-cell--${gameBoard[row][col]}`;
+    }
+  }
+
+  gameScore.innerText = score;
+}
+
+function setRandomCell() {
+  const [x, y] = [
+    getRandomNumber(),
+    getRandomNumber(),
+  ];
+
+  if (!gameBoard[x][y]) {
+    gameBoard[x][y] = getRandomNumber() === 1 ? 4 : 2;
```

Comment:
now probability is 50:50
because `Math.floor(Math.random() * 2)` generates `0` or `1`
if you need 10% probability better use 10
for example `Math.floor(Math.random() * 10)` will give you numbers from `0` to `9`